From 441c63641f9381d34bf18aa8fe17c6d5b80a1a80 Mon Sep 17 00:00:00 2001
From: aslammdkhan <aslam.khan@hpe.com>
Date: Thu, 26 May 2016 17:28:56 -0700
Subject: [PATCH] Changed the schema-to-idl auto generation script to populate
 value if it is present.

TG-907
Tags: fix, dev

The auto generation script is not looking for value part of key-value and is
not updating the value in set_subinterface idl api when min=0 and max=1.
Added code to check if value is present fill that also.

Change-Id: Ieb95c6f02bf8cc45430966606c8bdfe3781e429d
Signed-off-by: aslammdkhan <aslam.khan@hpe.com>
---
 ovsdb/ovsdb-idlc.in | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/ovsdb/ovsdb-idlc.in b/ovsdb/ovsdb-idlc.in
index 4bada77..ce81088 100755
--- a/ovsdb/ovsdb-idlc.in
+++ b/ovsdb/ovsdb-idlc.in
@@ -932,6 +932,8 @@ const struct ovsdb_datum *
                 txn_write_func = "ovsdb_idl_txn_write_clone"
             elif type.n_max == 1:
                 print "    union ovsdb_atom key;"
+                if type.value:
+                    print "    union ovsdb_atom value;"
                 print
                 print "    ovs_assert(inited);"
                 print "    if (%s) {" % nVar
@@ -942,7 +944,11 @@ const struct ovsdb_datum *
                 print "        datum.n = 0;"
                 print "        datum.keys = NULL;"
                 print "    }"
-                print "    datum.values = NULL;"
+                if type.value:
+                    print "    datum.values = &value;"
+                    print "    " + type.value.copyCValue("datum.values[0].%s" % type.value.type.to_string(), "%s[0]" % valueVar)
+                else:
+                    print "    datum.values = NULL;"
                 txn_write_func = "ovsdb_idl_txn_write_clone"
             else:
                 print "    size_t i;"
-- 
2.7.4

